hail/src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -351,8 +351,23 @@ object Interpret {
               assert(onKey)
               d.count { case (k, _) => elem.typ.ordering.lt(k, eValue) }
             case a: IndexedSeq[_] =>
-              assert(!onKey)
-              a.count(elem.typ.ordering.lt(_, eValue))
+              if (onKey) {
+                val (eltF, eltT) = orderedCollection.typ.asInstanceOf[TContainer].elementType match {
+                  case t: TBaseStruct => ( { (x: Any) =>
+                    val r = x.asInstanceOf[Row]
+                    if (r == null) null else r.get(0)
+                  }, t.types(0))
+                  case i: TInterval => ( { (x: Any) =>
+                    val i = x.asInstanceOf[Interval]
+                    if (i == null) null else i.start
+                  }, i.pointType)
+                }
+                val ordering = eltT.ordering
+                val lb = a.count(elem => ordering.lt(eltF(elem), eValue))
```

Comment:
Why aren't we using the fact that `a` is ordered? Couldn't this be `a.lowerBound(eValue, ordering.lt, eltF)`?

Reply:
Our interpreter and compiler currently have different orderings, I think (remember that nasty ordering bug from a while ago).